### DISCUSSION TITLE
Swarm support?

### DISCUSSION THREAD

**Discussion by adamshand** [louislam/dockge#433] [Feature Request]
### ⚠️ Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find similar feature request

### 🏷️ Feature Request Type

Other

### 🔖 Feature description

I'm wondering if there is any interest or intention in eventually supporting Swarm mode?

### ✔️ Solution

I've been experimenting with swarm and it's quite handy for automatically failing containers over to other nodes.  Useful for rolling upgrades, reboots, hardware failure etc.

**Comment by adamshand** [comment]
I guess that's a "no".  Oh well ...

**Comment by encryptblockr** [comment]
any plan to support docker swarm mode???

https://docs.docker.com/engine/swarm/swarm-mode/

**Comment by femabeh** [comment]
Sounds cool but a "Cluster View" would also be cool, so you can see how much notes and manager you have.

**Comment by ecker00** [comment]
Swarm mode would be great, currently managing it with cli and a few custom scripts, but using `dockge` would help a lot. Seems to be in popular demand: 

- https://github.com/louislam/dockge/issues/465
- https://github.com/louislam/dockge/issues/333

**Comment by suranyami** [comment]
Looking on the Insights/Network graph, it seems like no-one is working on this.

I'd definitely like to see this happen, so… forkers gonna fork…

**Comment by samerfarida** [comment]
Also looking for docker swarm support! Thank you

**Comment by tommy4st** [comment]
So, what is actually needed to support Docker Swarm?

Here some ideas:
- node discovery, see #540
- node monitoring
- "stack" deployment instead of or additional to "compose"

I think the most important of these is the deployment of "stacks", which is the core feature of Docker Swarm.

A main problem is, that all this is not compatible with podman, but when no swarm mode is enabled, you should hide this features anyway.

**Comment by gleitoncampos** [comment]
+1 for swarm support

I was really excited to use Dockge, but I needed to switch to Portainer because of the "stack deploy" feature

**Comment by rushowr** [comment]
+1 for Docker Swarm support

**Comment by dlbogdan** [comment]
+1 let's go! I'd love to help, might through in my elbow even, but with 2 small kids, someone else has to lead this.

**Comment by balgerion** [comment]
Is there any chance to have swarm support ? I’m using portaineer only because it have swarm.
Swarm doesn’t get enough love in community it’s literally 3 cmdlets and you have working HA in docker with standard docker compose support.

**Comment by githubbiswb** [comment]
+1 Docker Swarm support

**Comment by oceanfabreeze** [comment]
+1 For Swarm.

**Comment by Drallas** [comment]
+1 for Docker Swarm support

**Comment by Yankee707** [comment]
+1 for swarm!

**Comment by dfana01** [comment]
+1

**Comment by AndersPier** [comment]
+1 for Docker swarm support

**Comment by IndianaJoe1216** [comment]
+1 For Swarm support!

**Comment by Fifthdread** [comment]
I'd love swarm support. Dockge is a big thing I miss from migrating from Docker Compose to Docker Swarm. Specifically, being able to see all the swarm stacks, how they're doing, and do basics such as restart things or update even my swarm .yaml files, if possible.

**Reply by rushowr** [reply]
Exactly! Oh my kingdom for the time available to make the modifications myself! Would that I could, or rather that I had a kingdom to do so, I would in a heartbeat haha.

**Comment by finnk8** [comment]
+1 For Swarm support!

**Comment by log0u7** [comment]
+1 For Swarm support!

**Comment by acheslow** [comment]
Another +1

**Comment by zarix** [comment]
+1 For Swarm support!

**Comment by unrealer** [comment]
+1 for swarm support

**Comment by ivenos** [comment]
+1 for swarm support

**Comment by tuxpowered** [comment]
+100 swarm support and  also for managing remote docker instances.

**Comment by freeeflyer** [comment]
+1 swarm support as well..

**Comment by Iraithium** [comment]
+1 for Swarm support. Please, please.

**Comment by nkamenar** [comment]
+1 for me as well. I run all my containers exclusively through swarm these days.

**Comment by neilcitruslime** [comment]
I agree swarm support would be awesome

**Comment by brofids** [comment]
+1

**Comment by SamB-GB** [comment]
+1

**Comment by JensHKnudsen** [comment]
+1 for swarm

I Love, Love, Love Dockge... Only docker management tool that works properly on cell phones
Thanks

**Comment by StudioEtrange** [comment]
+1 :)
